core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1322,6 +1326,30 @@ class KafkaApis(val requestChannel: RequestChannel,
     txnCoordinator.handleInitPid(initPidRequest.transactionalId, sendResponseCallback)
   }
 
+  def handleEndTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleAbortTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleAddPartitionToTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleAddOffsetsToTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleWriteTxnMarkerRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleTxnOffsetCommitRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
```

Comment:
Could we send dummy responses instead of throwing an exception? KAFKA-4954 is trying to add a mechanism to delay the response of every client request due to request level throttling. It would be harder to do that if the implementation just throws an exception.